presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -149,27 +153,31 @@
     private final HiveCluster clientProvider;
     private final Function<Exception, Exception> exceptionMapper;
     private final boolean impersonationEnabled;
+    private final HiveMetastoreAuthentication metastoreAuthentication;
 
     @Inject
-    public ThriftHiveMetastore(HiveCluster hiveCluster, MetastoreClientConfig config)
+    public ThriftHiveMetastore(HiveCluster hiveCluster, MetastoreClientConfig config, HiveMetastoreAuthentication metastoreAuthentication)
     {
         this(
                 hiveCluster,
                 new ThriftHiveMetastoreStats(),
                 identity(),
-                requireNonNull(config, "config is null").isMetastoreImpersonationEnabled());
+                requireNonNull(config, "config is null").isMetastoreImpersonationEnabled(),
+                metastoreAuthentication);
```

Comment:
`requireNonNull(hiveClientConfig, "hiveClientConfig is null").getHiveMetastoreAuthenticationType());`